macros/variables/get_metric_tree.sql

```diff
@@ -13,19 +13,22 @@ are both upstream of Metric C AND not derived metrics themselves. #}
 {%- do metric_tree.update({'parent_set':[]}) -%}
 {# The derived set is a list of derived metrics. This includes all derived metrics referenced
 in the macro itself OR upstream of the metrics referenced in the macro #}
-{%- do metric_tree.update({'expression_set':[]}) -%}
+{%- do metric_tree.update({'derived_set':[]}) -%}
 {# The base set is the list of metrics that are provided into the macro #}
 {%- do metric_tree.update({'base_set':[]}) -%}
 {# The ordered derived set is the list of derived metrics that are ordered based on their
 node depth. So if Metric C were downstream of Metric A and B, which were also derived metrics,
 Metric C would have the value of 999 (max depth) and A and B would have 998, representing that they
 are one depth upstream #}
-{%- do metric_tree.update({'ordered_expression_set':{}}) -%}
+{%- do metric_tree.update({'ordered_derived_set':{}}) -%}
 
 {# {% if metric_list is iterable and (metric_list is not string and metric_list is not mapping) %}  #}
 {% set base_set_list = []%}
+{% set full_set_list = []%}
+{# {% do print(metric_list)%} #}
 {% for metric in metric_list %}
     {%- do base_set_list.append(metric.name) -%}
+    {# {%- set test_list = metric.full_metric_dependency() -%} #}
```

Comment:
This line right here is what is giving me issues. When I try and run this I get the following error message: `'dbt.contracts.graph.metrics.MetricReference object' has no attribute 'full_metric_dependency'`

And that is true! Because that attribute/helper function lives in `ResolvedMetricReference`. So I am not entirely sure how our compilation/parsing flows through macros that it is throwing errors here. 🤔